web/src/compose_fade_users.ts

```diff
@@ -1,36 +1,42 @@
+import type {BuddyUserInfo} from "./buddy_data";
 import * as compose_fade_helper from "./compose_fade_helper";
 import * as people from "./people";
 
-export type UserFadeConfig = {
-    get_user_id($li: JQuery): number;
-    fade($li: JQuery): void;
-    unfade($li: JQuery): void;
+type RowElement = JQuery | BuddyUserInfo;
```

Comment:
That’s an alarming type. Do we really mix these in with each other in the same array? Or is each caller only ever dealing with an array of one **or** an array of the other? If the latter, we need to use generics, not unions.

Reply:
Ah yeah it can be either, not both. Thanks for pointing this out, I wasn't quite sure how to do this.

I did the generic type for `UserFadeConfig`, but how do I incorporate that into functions? Ah okay I found something online, let me try and let me know if that looks right.